admin-dev/themes/default/js/bundle/product/product-combinations.js

```diff
@@ -54,16 +55,24 @@ var combinations = (function() {
       if (!tableRow.is('tr')) {
           throw new Error('Structure of table has changed, this function needs to be updated.');
       }
-      var priceImpactInput = tableRow.find('.attribute_priceTE').first();
-      var finalPriceLabel = tableRow.find('.attribute-finalprice span');
 
-      var impactOnPrice = Tools.parseFloatFromString(priceImpactInput.val());
-      var previousImpactOnPrice = Tools.parseFloatFromString(priceImpactInput.attr('value'));
+    // We need this because there is a specific data="smthg" attribute so we can't use data() function
+      var attributeId = tableRow.attr('data');
 
-      var currentFinalPrice = Tools.parseFloatFromString(finalPriceLabel.data('price'), true);
-      var finalPrice = currentFinalPrice - previousImpactOnPrice + impactOnPrice;
+      // Get combination final price value from combination form
+      var finalPrice = priceCalculation.getCombinationFinalPriceTaxExcludedById(attributeId);
+      var finalPriceLabel = tableRow.find('.attribute-finalprice span.final-price');
+      finalPriceLabel.html(finalPrice);
 
-      finalPriceLabel.html(Number(ps_round(finalPrice, 6)).toFixed(6));
+      // Update ecotax preview (tax included)
+      var combinationEcotaxTI = priceCalculation.getCombinationEcotaxTaxIncludedById(attributeId);
+      if (combinationEcotaxTI === 0) {
+        combinationEcotaxTI = priceCalculation.getProductEcotaxTaxIncluded();
+      }
+      var ecoTaxLabel = tableRow.find('.attribute-finalprice span.attribute-ecotax');
+      ecoTaxLabel.html(Number(ps_round(combinationEcotaxTI, 2)).toFixed(2)); // 2 digits for short
```

Comment:
Can't we use the priceDisplayPrecision here instead of the hardcoded `2`?

Reply:
No in this case the 2 is on purpose, this is displayed in the Combination list and using 6 decimals was kind of ugly (it's just a reminder in the list, when you edit the combination you get the full price with 6 decimals)

<img width="960" alt="Capture d’écran 2021-07-27 à 12 44 32" src="https://user-images.githubusercontent.com/13801017/127141467-6e56e19b-0bbe-44b1-8c42-202130c3de69.png">
(here because of the `Ecotax` label it would take two lines and it's really not ideal.. and this value is only for display)

<img width="1313" alt="Capture d’écran 2021-07-27 à 12 44 39" src="https://user-images.githubusercontent.com/13801017/127141523-65c61507-f0e7-4fa8-8fb0-d8bf2c9eb346.png">
(In the edition though we have more precision)

Reply:
I think the priceDisplayPrecision is 2 by default though, 6 is for compute precision I think

Reply:
Indeed, but actually, the display precision value used here is a hardcoded value of 6 that comes from:
https://github.com/PrestaShop/PrestaShop/blob/3d06748975e00c548e940bf82862d9d211ae94d4/src/PrestaShopBundle/Form/Admin/Type/CommonAbstractType.php#L37

I just fixed a bug related to this precision, I thought I could rely on computing precision but we reverted the changes and it's always 2, so it turns out in some cases the display precision is better (6) And in this case we absolutely need an accurate precision

So it's really badly done but I admit but I don't plan on fixing all the legacy inconsistencies related to precision in this PR... I already banged my head enough this morning because it made no sense.. Now it works that's all I care about for now 😅